OMCompiler/SimulationRuntime/c/meta/meta_modelica_data.h

```diff
@@ -138,15 +138,15 @@ typedef int mmc_switch_type;
 #define MMC_TAGPTR(p)             ((void*)((char*)(p) + 3))
 #define MMC_UNTAGPTR(x)           ((void*)((char*)(x) - 3))
 #define MMC_IS_INTEGER(X)         (0 == ((mmc_sint_t) (X) & 1))
-#define MMC_TAGFIXNUM(i)          (((i) << 1)+0)
+#define MMC_TAGFIXNUM(i)          ((((mmc_uint_t) (i)) << 1)+0)
```

Comment:
@atrosinenko Why use mmc_uint_t instead of mmc_sint_t ? Could they both be casted to mmc_uint_t?

Reply:
Unfortunately, I hardly remember right now. But this is probably as attempt to fix an UB (shifting the sign bit of -1 out of the number or something similar that the compiler complained about).